src/main/webapp/components/BuildSnapshot.js

```diff
@@ -1,14 +1,72 @@
-import * as React from "react";
+import * as React from 'react';
+import {getFields} from './utils/getFields.js';
 
 /**
  * Component used to render Build Information.
  *
  * @param {*} props input properties containing build information
  */
 export const BuildSnapshot = (props) => {
+  // State variable indicating whether the BuildSnapshot is open or closed.
+  // isOpen === true indicates the tray should be visible 
+  // and the indicator arrow on the Header should be facing downwards.
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const headerFields = ['description', 'commitHash', 'repository', 'status'];
+  const trayFields = ['builders', 'timestamp'];
+
+  const headerData = getFields(props.buildData, headerFields, '');
+  const trayData = getFields(props.buildData, trayFields, '');
+
+  return (
+    <div className = 'build-snapshot'>
+      <Header isOpen = {isOpen} onClick = {setIsOpen} data = {headerData}/>
+      <Tray isOpen = {isOpen} data = {trayData}/>
+    </div>
+  );
+}
+
+/**
+  * Subcomponent that holds the content displayed
+  * when the BuildSnapshot is collapsed.
+  *
+  * @param props an object containing a 'data' field
+  * which encapsulates the fields from headerFields.
+ */
```

Comment:
add 1 space to the `*/` to be in line with the rest of the documentation

Reply:
Space added here 8e1b25c